pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -285,6 +337,23 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo, provisionC
 		args = defaultArgs
 	}
 
+	// Add the volume to the spec and the mount to the daemon container
+	copyBinariesVolume, copyBinariesContainer := c.getCopyBinariesContainer()
+	if doBinaryCopyInit {
```

Comment:
It seems to me that it might be better to add volumes and init containers within the same `if` block to keep things a little more localized and keep the number of `if` checks lower. I would hope this would make this giant function a little easier to parse.

Reply:
Ok, I'll move that logic into the "main" `if` that activates this scenario if that's what you mean :)

Reply:
Btw I just moved this block, I did not create it.